src/datasources/balances-api/coingecko-api.service.ts

```diff
@@ -262,14 +286,26 @@ export class CoingeckoApi implements IPricesApi {
         ...args,
         tokenAddress,
       });
-      const cached = await this.cacheService.hGet(cacheDir);
-      const { key, field } = cacheDir;
-      if (cached != null) {
-        this.loggingService.debug({ type: 'cache_hit', key, field });
-        const cachedAssetPrice = AssetPriceSchema.parse(JSON.parse(cached));
-        result.push(cachedAssetPrice);
+      const memoryItem = await this.inMemoryCache.get<string>(
+        this.getMemoryKey(cacheDir),
+      );
+      if (memoryItem != null) {
+        this.logMemoryHit(cacheDir.key);
+        result.push(AssetPriceSchema.parse(JSON.parse(memoryItem)));
       } else {
-        this.loggingService.debug({ type: 'cache_miss', key, field });
+        this.logMemoryMiss(cacheDir.key);
+        const cacheItem = await this.cacheService.hGet(cacheDir);
+        if (cacheItem != null) {
+          this.logCacheHit(cacheDir);
+          await this.inMemoryCache.set(
+            this.getMemoryKey(cacheDir),
+            cacheItem,
+            this.pricesTtlSeconds * 1_000,
```

Comment:
If the value from Redis is has a remaining TTL of `1`, this will be cached for a further `this.pricesTtlSeconds * 1_000`. Is there any way to get the remaining TTL from Redis and set it as that? Otherwise we could doubling the TTL of these values.

Reply:
This value is in milliseconds, that's the reason we multiply it by 1000. The cache times should be the same for Redis and memory in this case. Will include a comment to document this!

Reply:
Added in [0cf8239](https://github.com/safe-global/safe-client-gateway/pull/2512/commits/0cf8239b6f15217eb2401d7ed840318c4b741771)